tests/modules/test_upload_contacts_from_file.py

```diff
@@ -1,237 +1,264 @@
+        self.assertEqual(parse_or_create_delay_num("18 928"),0)
+        self.assertEqual(parse_or_create_delay_num("18,928"),0)
+
+    def test_empty_previous_vaccination(self):
+        self.assertEqual(previous_vaccination(''), None)
+
+    def test_previous_vaccination_positive(self):
+        self.assertEqual(previous_vaccination("y"),True)
+        self.assertEqual(previous_vaccination("yes"),True)
+        self.assertEqual(previous_vaccination("yes  "),True)
+        self.assertEqual(previous_vaccination("yes asdfasfasf "),True)
+
+    def test_previous_vaccination_negative(self):
+        self.assertEqual(previous_vaccination("n"),False)
+        self.assertEqual(previous_vaccination("no"),False)
+        self.assertEqual(previous_vaccination("no  "),False)
+        self.assertEqual(previous_vaccination("no asdfasfasf "),False)
+
+
+    def test_fake_dates_for_parse_or_create_functional_dob(self):
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("40-15-2015", "40-15-2015", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("30-15-2015", "30-15-2015", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("12-35-2015", "12-35-2015", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("2014-20-15", "2014-20-15", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("12-00-2015", "12-00-2015", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("20000-20-15", "0000-20-15", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("2014-35-00", "2014-35-00", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("2016-12-40", "2016-12-40", 0)
+        with self.assertRaises(ValueError):
+            parse_or_create_functional_dob("2017-02-40", "2017-02-40", 0)
+
+    def test_nonexistent_dates_for_parse_or_create_functional_dob(self):
+        with self.assertRaises(TypeError):
+            parse_or_create_functional_dob("", "", 0)
+
+        delay1 = 0
+        delay2 = 5
+        dob1 = entered_date_string_to_date("10-15-2015")
+        parsed_func_dob = parse_or_create_functional_dob("", dob1, delay1)
+        parsed_func_dob2 = parse_or_create_functional_dob("", dob1, delay2)
+        
+        self.assertEqual(parsed_func_dob, datetime(2015, 10, 15).date())
+        self.assertEqual(parsed_func_dob2, datetime(2015, 10, 20).date())
+
+    def test_real_dates_for_parse_or_create_functional_dob(self):
+        funct_dob1 = "10-15-2015"
+        funct_dob2 = "10-18-2015"
+        dob1 = entered_date_string_to_date("10-15-2015")
+        delay1 = 0
+        delay2 = 3
+        parsed_func_dob1 = parse_or_create_functional_dob(funct_dob1, dob1, delay1)
+        parsed_func_dob2 = parse_or_create_functional_dob(funct_dob2, dob1, delay2)
```

Comment:
Can you pass the stuff instead of assigning to variables? You can always use 

```
parse_or_create_functional_dob(row_entry="10-15-2015", date_of_birth="10-15-2015", delay=0)
```

if the text isn't clear.